### PR TITLE
Intergate Cloudinary

### PR DESCRIPTION
**Pull request by shevchenkoviktoria**
When a POST request is made to the` /api/profile/upload` endpoint with a file attached, the multer middleware parses the file and stores it in memory.
The route handler then extracts the file data from `req.file`.buffer and uploads it to Cloudinary using `cloudinary.uploader.upload.`
Cloudinary accepts the image from the URL provided in the request and uploads it to the configured Cloudinary account.
Once the upload is successful, the Cloudinary response containing the upload result is received.
The URL of the uploaded image is extracted from the Cloudinary response `(uploadResult.url)`, and this URL is then used to add the photo to the database using` photoService.addPhoto.`
Finally, a response is sent back to the client indicating that the photo has been uploaded successfully, along with details such as the Cloudinary upload result and the URL of the uploaded photo.

### CODE REVIEW

File: routes/cloudinary-routes.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
+import cloudinary from '../services/cloudinary';
+
+function uploadAsset(req, res) {
```
Comment (NataliaDoudkina) [inline routes/cloudinary-routes.js:3]: This is not a route, it is a function, move it to the services or to the actual photo-routes file

File: services/cloudinary.js [hunk 2]
```diff
@@ -0,0 +1,11 @@
+import {v2 as cloudinary} from 'cloudinary';
+
+
+cloudinary.config({
```
Comment (NataliaDoudkina) [inline services/cloudinary.js:4]: This is a config file, not a service, move it to another folder named config or something where we can store all our config files

File: routes/cloudinary-routes.js [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+    const imageUrl = "https://upload.wikimedia.org/wikipedia/commons/a/ae/Olympic_flag.jpg";
```
Comment (NataliaDoudkina) [inline routes/cloudinary-routes.js:4]: What does this link do?
Did you test it from the frontend?

**Review by NataliaDoudkina — CHANGES_REQUESTED** [review]
Left some comments
We have some photo-routes that also need to be revisited and adjusted to use cloudinary,
has to be tested on FE

File: routes/cloudinary-routes.js (hunk 3 above)
Comment (shevchenkoviktoria) [reply routes/cloudinary-routes.js:4]: it is for uploading asset, i have not tested it

Comment (shevchenkoviktoria) [reply routes/cloudinary-routes.js:4]: image

File: config/cloudinary.config.js [hunk 4]
```diff
@@ -0,0 +1,10 @@
+const cloudinary = require('cloudinary').v2;
```
Comment (NataliaDoudkina) [inline config/cloudinary.config.js:1]: you can just leave a file name as cloudinary, "config" is redundant since it is already in the config folder.

**Review by NataliaDoudkina — COMMENTED** [review]
I don't see the file where you had the upload function, and I see that this PR is a new one?
Make sure this works with the frontend

File: config/cloudinary.config.js (hunk 4 above)
Comment (shevchenkoviktoria) [reply config/cloudinary.config.js]: fixed ✅

**Comment by shevchenkoviktoria** [comment]
> I don't see the file where you had the upload function, and I see that this PR is a new one? Make sure this works with the frontend

It is not new PR, I've just fixed it along the left comments. The upload function is ready. Frontend team can check it.

**Review by NataliaDoudkina — APPROVED** [review]
LGTM,
you can manually add the ENV variables to the Glitch account , I think it will not read them from the .env file

**Pull request merged**